services/src/test-library/src/main/java/org/openkilda/testing/service/lockkeeper/LockKeeperVirtualImpl.java

```diff
@@ -59,4 +59,34 @@ public void setController(Switch sw, String controller) {
         restTemplate.exchange(labService.getLab().getLabId() + "/lock-keeper/set-controller", HttpMethod.POST,
                 new HttpEntity<>(new SwitchModify(sw.getName(), controller), buildJsonHeaders()), String.class);
     }
+
+    @Override
+    public void blockAccessToIp(String ipAddress) {
+        throw new UnsupportedOperationException(
+                "blockAccessToIp method is not available on virtual env");
+    }
+
+    @Override
+    public void unblockAccessToIp(String ipAddress) {
+        throw new UnsupportedOperationException(
+                "unblockAccessToIp method is not available on virtual env");
+    }
+
+    @Override
+    public void blockAccessToPort(Integer port) {
+        throw new UnsupportedOperationException(
+                "blockAccessToPort method is not available on virtual env");
+    }
+
+    @Override
+    public void unblockAccessToPort(Integer port) {
+        throw new UnsupportedOperationException(
+                "unblockAccessToPort method is not available on virtual env");
+    }
+
+    @Override
+    public void allowAccessToEverything() {
+        throw new UnsupportedOperationException(
+                "allowAccessToEverything method is not available on virtual env");
+    }
```

Comment:
Not the best solution. We should keep virtual and hw implementations as close to each other as possible.
you can
1. to discuss: remove all the block/unblock methods from the interface at all.
2. Add 'hardware' implementation for `reviveSwitch` and `knockoutSwitch`, which will reduce the amount of UnsupportedOperationException-s (instead of increasing it)
3. add `knockoutFloodlight` to the interface. It will have a hardware implementation (through the lock keeper service) and we should think of how this can be implemented for virtual (use local docker, goto fl container and run the same iptables command?)

Reply:
it was discussed internally
1. decided to leave blockAccessToPort/unblockAccessToPort
2. done
3. decided not to add this method